test/e2e/density.go

```diff
@@ -186,6 +209,108 @@ var _ = Describe("Density", func() {
 			// Tune the threshold for allowed failures.
 			badEvents := BadEvents(events)
 			Expect(badEvents).NotTo(BeNumerically(">", int(math.Floor(0.01*float64(totalPods)))))
+
+			if itArg.runLatencyTest {
+				Logf("Schedling additional Pods to measure startup latencies")
+				// Use service to assure even spread of pods between nodes
+				service := &api.Service{
+					ObjectMeta: api.ObjectMeta{
+						Name: "latency-service",
+					},
+					Spec: api.ServiceSpec{
+						Selector: map[string]string{"name": additionalRCName},
+						Ports: []api.ServicePort{
+							{
+								Name:       "portname1",
+								Port:       80,
+								TargetPort: util.NewIntOrStringFromString("service1"),
+							},
+						},
+					},
+				}
+				_, err = c.Services(ns).Create(service)
+				expectNoError(err)
+
+				gotPods, err := c.Pods(ns).List(labels.Everything(), fields.Everything())
+				expectNoError(err)
+
+				var latencies []PodStartupLatencyData
+				additionalNameStr := strconv.Itoa(minionCount) + "-" + string(util.NewUUID())
+				additionalRCName = "my-hostname-latency" + additionalNameStr
+				config = RCConfig{Client: c,
+					Image:        "gcr.io/google_containers/pause:go",
+					Name:         additionalRCName,
+					Namespace:    ns,
+					PollInterval: itArg.interval,
+					Replicas:     minionCount,
+				}
+
+				observedStartTime := make(map[string]util.Time, 0)
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go func() {
+					defer GinkgoRecover()
+					watch, err := c.Pods(ns).Watch(labels.SelectorFromSet(labels.Set(map[string]string{"name": additionalRCName})), fields.Set{"status.phase": "Running"}.AsSelector(), gotPods.ResourceVersion)
```

Comment:
This benchmark is getting too complex already, I'd rather not another layer on top of watch to keep it simple. Worst that can happen is that one would need to rerun the test.